hack/verify-contextual-logging.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The KCP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o nounset
+set -o pipefail
+# Don't set errexit because logcheck's messages are in stderr
```

Comment:
`set -o errexit` exits the shell on a command failing, not output going to `stderr`, so you should be fine

Reply:
I was actually getting no output at all when that was set, for some reason. I didn't check `$?` after logcheck runs, perhaps it sets an error return code when it find non-conformat code.

Reply:
Double checked, `errexit` checks for non-zero status codes, and `logcheck` sets it to 3 when we run right now.

```shell
❯ logcheck -check-contextual ./pkg/...
[...]
❯ echo $?
3
```

Reply:
Yes, but inside of:

```sh
if ! logcheck ... ; then
```

It does not exit - let's chat when you've pushed your updates